server/src/main/java/org/elasticsearch/script/VectorScoreScriptUtils.java

```diff
@@ -46,31 +96,37 @@ public DenseVectorFunction(ScoreScript scoreScript, List<Number> queryVector, St
                 queryMagnitude += value * value;
             }
             queryMagnitude = Math.sqrt(queryMagnitude);
+            field.getElementType().checkVectorBounds(this.queryVector);
```

Comment:
So, for float this would add NaN and Infinite checking which I think would cause downstream issues anyway as you mentioned in the prior PR? If not I can change this to only operate on byte vectors. For byte it's the same bounds checking we do for ann queries.

Reply:
> So, for float this would add NaN and Infinite checking which I think would cause downstream issues anyway as you mentioned in the prior PR? 

Ah, I thought `checkVectorBounds` was also checking the dimensions. No worries here then.